### PR TITLE
Add jemalloc to docker images

### PR DESCRIPTION
**Pull request by DavidFair**
Adds jemalloc to the docker images as part of the migration from tcmalloc

### CODE REVIEW

**Review by martyngigg — COMMENTED** [review]
We might need to check how this is being implemented but for tcmalloc we needed the development package too as the CMake finders tend to only look for the unversioned `.so` files and not the versioned ones that the main packages install.

In [LinuxPackageScripts](https://github.com/mantidproject/mantid/blob/9fb364b5e538b1ebd9a11e5303079f47543b9692/buildconfig/CMake/LinuxPackageScripts.cmake#L168) we use the unversioned symlink to find the name of the major version to use for `LD_PRELOAD` so that the user package only has to depend on the main package and not the development package too.

File: docker-scripts/build_all.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# Build jenkins-node using previous images
+cd $ROOT_DIR/jenkins-node/docker-images/os-builder/
+./build.sh
+
+# Build static analysis images
+cd $ROOT_DIR/jenkins-node/docker-images/static-analysis/
+./build.sh
+
+cd $START_DIR
```
Comment (martyngigg) [inline docker-scripts/build_all.sh:21]: I'm not sure we need to switch back as the bash subprocess will exit leaving the parent cwd unchanged?

**Review by martyngigg — CHANGES_REQUESTED** [review]
Really, really minor and happy to be ignored. Shall we name the new scripts directory `scripts` rather than `docker-scripts`? All of the other directories with docker in their name seem to have docker files in them.

Should we update the top-level readme to say if you need to rebuild & push then use the scripts.

File: docker-scripts/build_all.sh (hunk 1 above)
Comment (DavidFair) [reply docker-scripts/build_all.sh]: Didn't know that, there was one script env (probably cmd) which dropped you in the last dir the script ran in, so I got into the habit of putting it in

**Comment by DavidFair** [comment]
Should be good to review again thanks

File: docker-scripts/build_all.sh (hunk 1 above)
Comment (martyngigg) [reply docker-scripts/build_all.sh]: Yeah you're right Windows cmd/bat scripts do that and apparently if you do `source script.sh` that will affect the current shell process too.

**Review by martyngigg — APPROVED** [review]
Looks good to go! :shipit:

**Pull request merged**